hardhat.config.js

```diff
@@ -115,7 +114,7 @@ module.exports = {
   },
   networks: {
     hardhat: {
-      allowUnlimitedContractSize,
+      allowUnlimitedContractSize: !withOptimizations,
```

Comment:
any reason for this rollback ? IMO, if the coverage is enabled, then optimizations should be on, but contract size should not be limitted. (coverage does increasse the size)